bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/RuleRegistryImpl.java

```diff
@@ -57,6 +67,14 @@ protected void addProvider(Provider<Rule> provider) {
     public synchronized void add(Rule element) {
         super.add(element);
         addIntoRuleEngine(element);
+        notify(RuleEventFactory.createRuleAddedEvent(element, SOURCE));
```

Comment:
I think the enum was only used for the listeners. But you can change it to be protected.